packages/developer-console-spa/src/components/ConfigureSearch/Steps/Step1.tsx

```diff
@@ -0,0 +1,131 @@
+import { yupResolver } from '@hookform/resolvers/yup';
+import { CodeEditor, Language } from '@patternfly/react-code-editor';
+import { ActionGroup, Button, Form, FormGroup, FormSection, FormSelect, FormSelectOption, Grid, GridItem, TextInput } from '@patternfly/react-core';
+import { useStateMachine } from 'little-state-machine';
+import { useCallback, useMemo } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import saveState from './store';
+import * as yup from 'yup';
+
+type Step1 = Pick<App.SearchConfig, 'method' | 'apiUrl' | 'body'>
+
+const formSchema = yup.object().shape( {
+  method: yup.string().required(),
+  apiUrl: yup.string()
+    .url( 'API URL must be a valid url' )
+    .required( 'API URL is mandatory' ),
+  body: yup.string(),
+});
+
+export default function ConfigureSearchStep1 ( { onNext, onReset }: IConfigureSearchStepProps ) {
+  const { control, handleSubmit, watch, formState: { errors, isValid } } = useForm<Step1>( {
+    mode: 'onBlur',
+    resolver: yupResolver(formSchema),
+  });
+  const { actions, state } = useStateMachine( { nextStep: saveState } );
+
+  const formData = state.formData;
+
+  const method = watch('method');
+  const httpMethods = [
+    { label: 'GET', value: 'GET', isDisabled: false },
+    { label: 'POST', value: 'POST', isDisabled: false },
+  ];
+
+  const isRequestBodyVisible = useMemo(() => {
```

Comment:
As this is a simple tertiary check memo can be avoided and on fly computation is better

Reply:
I'm using useMemo because that way, it only changes when the method or formData.method changes. So it does not have to run the checks on every render.